src/tike/ptycho/exitwave.py

```diff
@@ -0,0 +1,204 @@
+#     denom = abs2_Psi * cp.square( xi_alpha_minus_one ) + I_e - abs2_Psi
+
+#     rhs = cp.sum( xi_abs_Psi2 * ( numer / denom ), axis = (-1, -2 ) )   
+
+#     f_eq_0 = cp.abs( lhs - rhs )
+
+#     II = cp.argmin( f_eq_0, axis = 0 )
+
+#     step_length = cp.zeros( ( abs2_Psi.shape[0], abs2_Psi.shape[1] ), dtype = 'float32' )      
+
+#     for pp in cp.arange(0, Nscpm ):
+#         step_length[ pp, : ] = step_length_test[ II[ pp, : ], pp ]
+
+#     return step_length, f_eq_0
+  
+###################################################################################################
+
+# if __name__ == "__main__":
+
+    # import h5py
+
+    # f = h5py.File( '/net/s8iddata/export/8-id-ECA/Analysis/atripath/Python/poisson_exwv/pythonTIKE_poisson_testing_31May2023.mat', 'r' )
+
+    # Nspos = int( np.ndarray.item( np.array( f['Nspos'] )))    
+    # Nscpm = int( np.ndarray.item( np.array( f['Nscpm'] ))) 
+
+    # sz_Nr = 256
+    # sz_Nc = 384
+
+    # xi       = cp.array( f['xi'] ).reshape( ( Nspos, sz_Nr, sz_Nc ), order='F')           # = 1 - I_m / I_e
+    # abs2_Psi = cp.array( f['abs2_Psi'] ).reshape( ( Nscpm, Nspos, sz_Nr, sz_Nc ), order='F')
+    # I_e      = cp.array( f['I_e'] ).reshape( ( Nspos, sz_Nr, sz_Nc ), order='F')
+    # I_m      = cp.array( f['I_m'] ).reshape( ( Nspos, sz_Nr, sz_Nc ), order='F')
+
+    # measured_pixels = cp.array( f['not_I_m_eq0'] ).reshape( ( sz_Nr, sz_Nc ), order='F')
+    # alpha_test      = cp.squeeze( cp.array( f['alpha_test'] ))
+ 
+    # #=====
+
+    # poiss_steplength_exact, f_eq_0 = poisson_steplength_exact( xi, abs2_Psi, I_e, I_m, measured_pixels, alpha_test )
+    # poiss_steplength_approx        = poisson_steplength_approx( xi, abs2_Psi, I_e, I_m, measured_pixels, 0.50 * cp.ones( ( Nscpm, Nspos )), 0.70 )
+    # poiss_steplength_ptychoshelves = poisson_steplength_ptychoshelves( xi, I_e, I_m, measured_pixels, 0.5 * cp.ones( Nspos ), 0.50 )
+
+    # #=====
+
+    # poiss_steplength_exact         = cp.asnumpy( poiss_steplength_exact )
+    # poiss_steplength_approx        = cp.asnumpy( poiss_steplength_approx )
+    # poiss_steplength_ptychoshelves = cp.asnumpy( poiss_steplength_ptychoshelves )
+    # alpha_test                     = cp.asnumpy( alpha_test )
+    # f_eq_0                         = cp.asnumpy( f_eq_0 )
+
+    # import matplotlib.pyplot as plt
+
+    # for pp in np.arange( 0, Nscpm ):
+    #     plt.subplot( 1, Nscpm, pp + 1 )
+    #     ax = plt.gca()
+    #     ax.plot( poiss_steplength_exact[ pp, : ], 0.5 + np.arange(0,Nspos), marker='x', color="red")
+    #     ax.plot(  np.squeeze( poiss_steplength_ptychoshelves ), 0.5 + np.arange(0,Nspos), marker='.', color="blue")
+    #     ax.plot(  poiss_steplength_approx[ pp, : ], 0.5 + np.arange(0,Nspos), marker='.', color="white")
+    #     ax.imshow( np.log10(f_eq_0[ :, pp, : ].T), interpolation='none', extent = [ np.min( alpha_test[:,pp] ), np.max( alpha_test[:,pp] ), Nspos, 0 ], aspect="auto" )
+        
```

Comment:
Dead code should not be included in the pull request. If you're still developing this feature, you can store it on a personal branch and include it in a future pull request.